src/transport_ofi.c

```diff
@@ -559,8 +572,16 @@ static inline int atomic_limitations_check(void)
         ret = fi_atomicvalid(shmem_transport_ofi_epfd, SHM_DT_INT[i], SHM_BOPS[j],
                         &atomic_size);
         if(ret!=0 || atomic_size == 0) {
-           OFI_ERRMSG("ret=%d atomic_size=%d %d %d\n", ret, (int)atomic_size, i, j);
-	   return ret;
+            fprintf(stderr, "Atomic Support: data size %d not "\
+                    "supported with %s ", SHMEM_Dtsize[SHM_DT_INT[i]],
+                    SHM_NAMED_BOPS[j]);
+            if(atomicwarn) {
+                fprintf(stderr, "WARNING\n");
```

Comment:
The output needs a little work:

>  Atomic Support: data size 8 not supported with SUM
>  WARNING

Should be more like:

> Warning: OFI detected no support for atomic 'MAX' on type 'double'

You can create a format string to handle both cases like this:

```"%s: OFI detected no support...", atomicward ? "Warning" : "Error", ...```

And then abort when ```atomicwarn``` is not set.